### PR TITLE
Yaroslav bulin

### PR DESCRIPTION
**Pull request by slavabulin**

### CODE REVIEW

File: Homework2/Task1Tests/BitInserterTests.cs [hunk 1]
```diff
@@ -53,13 +53,29 @@ public void InsertBitsOnesFromZeroIndexTo15IndexShouldReturnMinus65536()
...
             retVal = bi.InsertBits(firstNumber, secondNumber, firstIndex, lastIndex);
 
             //assert
             Assert.AreEqual(-65536, retVal);
         }
+        [TestMethod]
+        public void InsertBitsBothIndexesAreEqualShouldReturnNull()
+        {
+            //arrange
+            int firstIndex = 0;
```
Comment (IliaOrlov) [inline Homework2/Task1Tests/BitInserterTests.cs]: `firstIndex` is zero in all tests
please add tests with different values for `firstIndex`

File: Homework2/Task1/Task1.cs [hunk 2]
```diff
@@ -27,12 +27,13 @@ public class BitInserter
...
-        public int InsertBits(int firstNumber, int secondNumber, int firstIndex, int lastIndex)
+        public int? InsertBits(int firstNumber, int secondNumber, int firstIndex, int lastIndex)
         {
             uint insertTo, insertFrom;
             if (lastIndex < firstIndex
-                || lastIndex > 32
-                || firstIndex < 0) return 0;
+                || lastIndex > 31
+                || firstIndex < 0
+                || firstIndex == lastIndex) return null;
```
Comment (IliaOrlov) [inline Homework2/Task1/Task1.cs]: add tests to chech this conditions

File: Homework2/Task2.Tests/MaximumFinderTests.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            //arrange
+            int[] inputArray = new int[] { Int32.MinValue, 0, Int32.MaxValue};
+            int retVal;
+            int expected = Int32.MaxValue;
+            //act
+            retVal = mf.FindMax(inputArray);
+            //assert
+            Assert.AreEqual(expected, retVal);
+        }
+    }
```
Comment (IliaOrlov) [inline Homework2/Task2.Tests/MaximumFinderTests.cs]: add test for case when inputArray is null

File: Homework2/Task3.Tests/SumCheckerTests.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            //arrange
+            int[] inputArray = new int[] {11, 0, 123458, -1};
+            int expectedVal = -1;
+            int retVal;
+            //act
+            retVal = sch.Check(inputArray);
+            //assert
+            Assert.AreEqual(expectedVal, retVal);
+        }
+    }
```
Comment (IliaOrlov) [inline Homework2/Task3.Tests/SumCheckerTests.cs]: add null argument test

File: Homework2/Task4Tests/StringConcatenatorTests.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            //arrange
+            string str1 = String.Empty;
+            string str2 = String.Empty;
+            string rezStr;
+            //act
+            rezStr = sc.Concatenate(str1, str2);
+            //assert
+            Assert.AreEqual("", rezStr);
+        }
+    }
```
Comment (IliaOrlov) [inline Homework2/Task4Tests/StringConcatenatorTests.cs]: null in parameters

File: Homework3/Task2/Sorter.cs [hunk 6]
```diff
@@ -50,6 +50,7 @@ public static void SortRowsByKey(ref int[,] inputArray, int[] key, Order order =
         }
         public static int[] GetKey(int[,]inputArray, Order order)
```
Comment (IliaOrlov) [inline Homework3/Task2/Sorter.cs]: misleading function name
the function returns Min or Max from each row, but it's not obvious

File: Homework4/Task2.Tests/EuclidAlgorithmTests.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+{
+    [TestClass]
+    public class EuclidAlgorithmTests
+    {
+        [TestMethod]
+        public void TestMethod1()
+        {
+        }
+    }
+}
```
Comment (IliaOrlov) [inline Homework4/Task2.Tests/EuclidAlgorithmTests.cs]: no tests?

File: Homework2/Task2.Tests/MaximumFinderTests.cs (hunk 3 above)
Comment (IliaOrlov) [reply Homework2/Task2.Tests/MaximumFinderTests.cs]: wrap FindMax method, hide internal parameters with default values
otherwise you need test them
